fastlane/Fastfile

```diff
@@ -1,26 +1,41 @@
 # frozen_string_literal: true
 
-default_platform(:ios)
+default_platform(:mac)
 fastlane_require 'dotenv'
 fastlane_require 'shellwords' # used for shell-escaping paths
 
+# Use this instead of getting values from ENV directly
+# It will throw an error if the requested value is missing
+def get_required_env(key)
+  UI.user_error!("Environment variable '#{key}' is not set. Have you setup #{USER_ENV_FILE_PATH} correctly?") unless ENV.key?(key)
+  ENV.fetch(key)
+end
+
 # Constants
 USER_ENV_FILE_PATH = File.join(Dir.home, '.simplenotemacos-env.default')
 PROJECT_FOLDER = Pathname.new(File.join(Dir.pwd, '..')).expand_path.to_s
 WORKSPACE = 'Simplenote.xcworkspace'
 INTERNAL_SCHEME = 'Simplenote'
 APP_STORE_SCHEME = 'Simplenote-AppStore'
 BUILD_FOLDER = 'build'
-APP_STORE_CONNECT_API_KEY_PATH = File.join(Dir.home, '.configure', 'simplenote-macos', 'secrets', 'app_store_connect_fastlane_api_key.json')
 APP_STORE_BUNDLE_IDENTIFIER = 'com.automattic.SimplenoteMac'
 VERSION_FILE_PATH = File.join(PROJECT_FOLDER, 'config', 'Version.Public.xcconfig')
 DEFAULT_BRANCH = 'trunk'
 GITHUB_REPO = 'Automattic/simplenote-macos'
 
+Dotenv.load(USER_ENV_FILE_PATH)
+
+APPLE_TEAM_ID = 'PZYM8XX95Q'
+# Generate the App Store Connect API key to download code signing certificates and profiles with `app_store_code_signing`
+APPLE_API_KEY = app_store_connect_api_key(
+  key_id: get_required_env('APP_STORE_CONNECT_API_PZYM8XX95Q_KEY_ID'),
+  issuer_id: get_required_env('APP_STORE_CONNECT_API_PZYM8XX95Q_ISSUER_ID'),
+  key_content: get_required_env('APP_STORE_CONNECT_API_PZYM8XX95Q_KEY_CONTENT')
+)
```

Comment:
Putting these lines here in the global scope not only means that every invocation of fastlane will call the `app_store_connect_api_key` method even when we don't need it for the lane we're running (and even when running `fastlane lanes` to list available lanes), but also that any run of `fastlane` will fail if you don't have one of the `get_required_env` vars set in your environment, while I can totally imagine devs not having those env vars set local and wanting to run a lane which doesn't need, yet won't be able to because of this global call.

---

I'd suggest instead to:
 - Either create a helper method to do those lines and call it only in the lanes that need it
 - Or even better, map the environment variables used by `app_store_connect_api_key` action's `ConfigItem` with the env vars used here (`SPACESHIP_CONNECT_API_KEY_ID="${APP_STORE_CONNECT_API_PZYM8XX95Q_KEY_ID}"` etc)—either in the Fastfile via `ENV['…']="…"`, or arguably better, in `~/.mobile-secrets/…/simplenote-macos/environment`—so that you can then just call the `app_store_connect_api_key` without any parameter in the lanes you need the ASC API key (thus no need to "pollute" the Fastfile to create a dedicated helper method when the action w/o params already does what you need)

Reply:
In addition, calling `app_store_connect_api_key` action only within the lane that needs it means that you also don't need to store the result of the action in a variable to use it afterwards, because `app_store_connect_api_key` sets the key as a shared value in the lane context, which can then be automatically picked up by subsequent actions that need an API Key 😉 

See https://docs.fastlane.tools/app-store-connect-api/#use-the-shared-value-in-lane-context

Reply:
Great idea! I implemented this in https://github.com/Automattic/simplenote-macos/commit/1e06f78a2ebe3afe668911f87872af42e244894c